### PR TITLE
Use sqldelight in migration

### PR DESCRIPTION
**Pull request by AntsyLich**

### CODE REVIEW

File: app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package eu.kanade.data.category
+
+import eu.kanade.data.DatabaseHandler
+import eu.kanade.domain.category.repository.MangaCategoryRepository
+
+class MangaCategoryRepositoryImpl(
+    private val handler: DatabaseHandler,
+) : MangaCategoryRepository {
+
+    override suspend fun moveMangaToCategories(mangaId: Long, categoryIds: List<Long>) {
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt:10]: Could probably be in [MangaRepository]

File: app/src/main/java/eu/kanade/data/track/TrackRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import eu.kanade.data.DatabaseHandler
+import eu.kanade.domain.track.model.Track
+import eu.kanade.domain.track.repository.TrackRepository
+
+class TrackRepositoryImpl(
+    private val handler: DatabaseHandler,
+) : TrackRepository {
+
+    override suspend fun getMangaTracks(mangaId: Long): List<Track> {
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/data/track/TrackRepositoryImpl.kt:11]:
```suggestion
    override suspend fun getTracksByMangaId(mangaId: Long): List<Track> {
```

File: app/src/main/java/eu/kanade/domain/category/interactor/GetCategories.kt [hunk 3]
```diff
@@ -11,4 +11,8 @@ class GetCategories(
     fun subscribe(): Flow<List<Category>> {
         return categoryRepository.getAll()
     }
+
+    suspend fun awaitWithMangaId(mangaId: Long): List<Category> {
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/domain/category/interactor/GetCategories.kt:15]:
```suggestion
    suspend fun await(mangaId: Long): List<Category> {
```

File: app/src/main/java/eu/kanade/domain/manga/interactor/GetMangaAndChapters.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import eu.kanade.domain.chapter.model.Chapter
+import eu.kanade.domain.chapter.repository.ChapterRepository
+import eu.kanade.domain.manga.model.Manga
+import eu.kanade.domain.manga.repository.MangaRepository
+import eu.kanade.tachiyomi.util.system.logcat
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import logcat.LogPriority
+
+class GetMangaAndChapters(
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/domain/manga/interactor/GetMangaAndChapters.kt:12]:
```suggestion
class GetMangaWithChapters(
```

File: app/src/main/java/eu/kanade/domain/track/interactor/GetTracksForManga.kt [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package eu.kanade.domain.track.interactor
+
+import eu.kanade.domain.track.model.Track
+import eu.kanade.domain.track.repository.TrackRepository
+import eu.kanade.tachiyomi.util.system.logcat
+import logcat.LogPriority
+
+class GetTracksForManga(
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/domain/track/interactor/GetTracksForManga.kt:8]:
```suggestion
class GetTracks(
```

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaPresenter.kt [hunk 6]
```diff
@@ -158,9 +158,10 @@ class MangaPresenter(
         // Keeps subscribed to changes and sends the list of chapters to the relay.
         presenterScope.launchIO {
             manga.id?.let { mangaId ->
-                getChapterByMangaId.subscribe(mangaId)
-                    .collectLatest { domainChapters ->
-                        val chapterItems = domainChapters.map { it.toDbChapter().toModel() }
+                getMangaWithChapters.subscribe(mangaId)
+                    .collectLatest { (_, chapters) ->
+                        // Chapters
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaPresenter.kt:163]: Is this comment needed?

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaPresenter.kt [hunk 7]
```diff
@@ -158,9 +158,10 @@ class MangaPresenter(
         // Keeps subscribed to changes and sends the list of chapters to the relay.
         presenterScope.launchIO {
             manga.id?.let { mangaId ->
-                getChapterByMangaId.subscribe(mangaId)
-                    .collectLatest { domainChapters ->
-                        val chapterItems = domainChapters.map { it.toDbChapter().toModel() }
+                getMangaWithChapters.subscribe(mangaId)
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaPresenter.kt:161]: Wouldn't it be more worth it to use [GetChapterByMangaId] as manga ain't used?

File: app/src/main/java/eu/kanade/tachiyomi/data/track/komga/Komga.kt [hunk 8]
```diff
@@ -105,12 +107,12 @@ class Komga(private val context: Context, id: Int) : TrackService(id), EnhancedT
             null
         }
 
-    override fun isTrackFrom(track: Track, manga: Manga, source: Source?): Boolean =
-        track.tracking_url == manga.url && source?.let { accept(it) } == true
+    override fun isTrackFrom(track: DomainTrack, manga: DomainManga, source: Source?): Boolean =
+        track.remoteUrl == manga.url && source?.let { accept(it) } == true
 
-    override fun migrateTrack(track: Track, manga: Manga, newSource: Source): Track? =
+    override fun migrateTrack(track: DomainTrack, manga: DomainManga, newSource: Source): eu.kanade.domain.track.model.Track? =
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/tachiyomi/data/track/komga/Komga.kt:113]: ?
```suggestion
    override fun migrateTrack(track: DomainTrack, manga: DomainManga, newSource: Source): DomainTrack? =
```

File: app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt (hunk 1 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt:10]: probably but similarly getChaptersForManga also technically qualifies for [MangaRepository]

File: app/src/main/java/eu/kanade/tachiyomi/data/track/komga/Komga.kt (hunk 8 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/data/track/komga/Komga.kt:113]: Missed it

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaPresenter.kt (hunk 7 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaPresenter.kt:161]: Gonna use it in a follow up pr anyway so just kept it

File: app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt (hunk 1 above)
Comment (ghostbear) [reply app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt:10]: It's more about is it worth having another Repository for a table that is just relations

Comment (AntsyLich) [reply app/src/main/java/eu/kanade/data/category/MangaCategoryRepositoryImpl.kt:10]: Good point

**Pull request merged**